### PR TITLE
Implements more methods on Python Response API and adds FormData API.

### PR DESCRIPTION
**Pull request by dom96**
The FormData API does not have a native Python equivalent, but it is a very useful API in Workers. This PR implements it and ties it into the Response API implementation.

### CODE REVIEW

File: .ruff.toml [hunk 1]
```diff
@@ -27,6 +27,7 @@ ignore = [
   "PLR2004", # Magic value used in comparison
   "TRY003", # Avoid specifying long messages outside the exception class
   "UP038",  # Use X | Y in isinstance check instead of (X, Y)
+  "PLR0911", # too many return statements
```
Comment (danlapid) [inline .ruff.toml:30]: Can we inline this ignore to the specific statement where we want to ignore this error type?

Comment (dom96) [reply .ruff.toml:30]: we can, but I think we can ignore it everywhere in the same way we are ignoring "too many branches"

File: src/pyodide/internal/workers.py [hunk 2]
```diff
@@ -71,8 +116,78 @@ def __init__(
...
+            )
+        except JsException as exc:
+            raise _to_python_exception(exc) from exc
+
+
+# TODO: Implement pythonic blob API
+FormDataValue = "str | js.Blob"
+
+
+class FormData:
```
Comment (hoodmane) [inline src/pyodide/internal/workers.py:156]: Maybe this should subclass `abc.MutableMapping[str, FormDataValue]`?

**Review by hoodmane — APPROVED** [review]
Looks good to me. I left some minor comments.

File: src/workerd/server/tests/python/sdk/worker.py [hunk 3]
```diff
@@ -1,8 +1,12 @@
...
-from cloudflare.workers import Response, fetch
+import js
 
+from cloudflare.workers import FormData, Response, fetch
 
+
+# Each path in this handler is its own test. The URLs that are being fetched
+# here are defined in server.py.
 async def on_fetch(request):
     if request.url.endswith("/modify"):
```
Comment (hoodmane) [inline src/workerd/server/tests/python/sdk/worker.py:11]: Would be nice to refactor this to use a dispatch table and a decorator.

Comment (hoodmane) [reply src/workerd/server/tests/python/sdk/worker.py:11]: Maybe I'll make a followup PR to rearrange this set of tests into a tidier form.

File: src/pyodide/internal/workers.py [hunk 4]
```diff
@@ -71,8 +116,78 @@ def __init__(
...
+    def __init__(self, form_data: "js.FormData | None | dict[str, FormDataValue]"):
+        if form_data:
+            if isinstance(form_data, dict):
+                self.js_form_data = js.FormData.new()
+                for item in form_data.items():
+                    self.js_form_data.append(item[0], item[1])
```
[the remark this answers is not in the document]
Comment (dom96) [reply src/pyodide/internal/workers.py:162]: it does not

[the remark this answers is not in the document]
Comment (hoodmane) [reply src/pyodide/internal/workers.py:162]: Ah indeed looks like the `FormData` constructor just does not work that way.

**Pull request merged**